### PR TITLE
Integrate DAS column subnets to peer scoring 

### PR DESCRIPTION
**Pull request by Nashatyrev**
## PR Description

Integrate DAS column subnets to peer scoring

### CODE REVIEW

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/Eth2P2PNetworkBuilder.java [hunk 1]
```diff
@@ -356,14 +369,58 @@ protected DiscoveryNetwork<?> buildNetwork(
...
         new TargetPeerRange(
             discoConfig.getMinPeers(),
             discoConfig.getMaxPeers(),
             discoConfig.getMinRandomlySelectedPeers());
+    final Supplier<SpecVersion> currentSpecVersionSupplier =
+        () -> combinedChainDataClient.getRecentChainData().getCurrentSpec();
     final SchemaDefinitionsSupplier currentSchemaDefinitions =
         () -> combinedChainDataClient.getRecentChainData().getCurrentSpec().getSchemaDefinitions();
+    final Supplier<Optional<UInt64>> currentSlotSupplier =
+        () -> combinedChainDataClient.getRecentChainData().getCurrentSlot();
```
Comment (zilm13) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/Eth2P2PNetworkBuilder.java:386]: nit: could consider to move it to separate class with `combinedChainDataClient` input

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/Eth2P2PNetworkBuilder.java [hunk 2]
```diff
@@ -356,14 +369,58 @@ protected DiscoveryNetwork<?> buildNetwork(
...
+
+    final PeerSubnetSubscriptions.NodeIdToDataColumnSidecarSubnetsCalculator
+        nodeIdToDataColumnSidecarSubnetsCalculator =
+            (nodeId, extraSubnetCount) -> {
+              SpecVersion currentSpecVersion = currentSpecVersionSupplier.get();
+              Integer custodyRequirement =
```
Comment (zilm13) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/Eth2P2PNetworkBuilder.java:392]: maybe finish earlier if we are preElectra?
Though we may want to have some of them in a few epochs before the fork.

File: networking/eth2/src/testFixtures/java/tech/pegasys/teku/networking/eth2/peers/StubPeerScorer.java [hunk 3]
```diff
@@ -32,7 +39,12 @@ public void setScore(
       final SszBitvector attestationSubscriptions,
       final SszBitvector syncCommitteeSubnetSubscriptions,
       final int score) {
-    candidateScores.put(Pair.of(attestationSubscriptions, syncCommitteeSubnetSubscriptions), score);
+    candidateScores.put(
+        new SubnetSubscriptionsKey(
+            attestationSubscriptions,
+            syncCommitteeSubnetSubscriptions,
+            SszBitvectorSchema.create(128).getDefault()),
```
Comment (zilm13) [inline networking/eth2/src/testFixtures/java/tech/pegasys/teku/networking/eth2/peers/StubPeerScorer.java:46]: Could you please detach it to const and use here and below

**Review by zilm13 — APPROVED** [review]
LGTM, just few nits

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/Eth2P2PNetworkBuilder.java (hunk 1 above)
Comment (zilm13) [reply networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/Eth2P2PNetworkBuilder.java:386]: whole calculator i mean

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/Eth2P2PNetworkBuilder.java (hunk 2 above)
Comment (Nashatyrev) [reply networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/Eth2P2PNetworkBuilder.java:392]: This one was refactored and gives up earlier now

**Pull request merged**